src/Features/CSharp/Portable/Braces/CSharpBracesDiagnosticAnalyzer.cs

```diff
@@ -0,0 +1,404 @@
+// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using Microsoft.CodeAnalysis.Diagnostics;
+using Microsoft.CodeAnalysis.CSharp.CodeStyle;
+using System.Collections.Generic;
+
+namespace Microsoft.CodeAnalysis.CSharp.Diagnostics.AddBraces
+{
+    [DiagnosticAnalyzer(LanguageNames.CSharp)]
+    internal sealed class CSharpAddBracesDiagnosticAnalyzer : DiagnosticAnalyzer, IBuiltInAnalyzer
+    {
+        private static readonly LocalizableString s_addBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Add_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_addBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Add_braces_to_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableTitle =
+            new LocalizableResourceString(nameof(FeaturesResources.Remove_braces), FeaturesResources.ResourceManager,
+                typeof(FeaturesResources));
+
+        private static readonly LocalizableString s_removeBracesLocalizableMessage =
+            new LocalizableResourceString(nameof(WorkspacesResources.Remove_braces_from_0_statement), WorkspacesResources.ResourceManager,
+                typeof(WorkspacesResources));
```

Comment:
why is one message using typeof(FeatureResources) but another is using typeof(WOrkspaceResources)?